knotx-knots/knotx-knot-action/src/main/java/com/cognifide/knotx/knot/action/ActionKnotVerticle.java

```diff
@@ -279,18 +298,20 @@ private void clearFromActionAttributes(Element item) {
         .forEach(attr -> item.removeAttr(attr.getKey()));
   }
 
-  private void addHiddenInputTag(Element form, String fragmentIdentifier) {
-    Matcher matcher = ACTION_FRAGMENT_IDENTIFIER_PATTERN.matcher(fragmentIdentifier);
-    if (matcher.find()) {
-      String formIdentifier = matcher.group(2);
-
-      Attributes attributes = Stream.of(
-          new Attribute("type", "hidden"),
-          new Attribute("name", configuration.formIdentifierName()),
-          new Attribute("value", StringUtils.isNotBlank(formIdentifier) ? formIdentifier : DEFAULT_FORM_IDENTIFIER))
-          .collect(Attributes::new, Attributes::put, Attributes::addAll);
-      form.prependChild(new Element(Tag.valueOf("input"), "/", attributes));
-    }
+  private void addHiddenInputTag(Element form, List<String> fragmentIdentifiers) {
+    fragmentIdentifiers.stream().filter(id -> id.startsWith(ACTION_FRAGMENT_IDENTIFIER)).findFirst().ifPresent(fragmentIdentifier -> {
```

Comment:
make idents with this stream invocation to enable better readability, e.g.:
```java
fragmentIdentifiers.stream()
               .filter(id -> id.startsWith(ACTION_FRAGMENT_IDENTIFIER))
               .findFirst()
               .ifPresent(fragmentIdentifier -> {
                        //body...
               });
```
Additionally, the `body` I would put to separate method so at the end of the day the whole `addHiddenInputTag` method would be more readable..